openshift-maven-plugin: service.yml fragment with ports creates service with unnamed port mapping

### Description

Using openshift-maven-plugin in a Spring Boot app, with the "spring boot" generator.

In the absence of any fragments the project builds and deploys successfully to OpenShift.  Once service is defined, "http" TCP 80 -> 8080

The intention of the fragment is to define an ADDITIONAL service, "hazelcast" TCP 5701 -> 5701.

src/main/jkube/service.yml:
```
spec:
  ports:
    - name: hazelcast
      protocol: TCP
      port: 5701
      targetPort: 5701
```

With the above fragment present, openshift-maven-plugin generates:

target/jkube/applyJson/namespace/service-xxx.json

including the following extract:

```
  "spec" : {
    "clusterIP" : "172.30.157.230",
    "ports" : [ {
      "port" : 5701,
      "protocol" : "TCP",
      "targetPort" : 5701
    } ],
```

Note that the port entry has no name.

At this point deployment is successful; TCP 5701 -> 5701 is exposed by an unnamed port mapping in service-xxx, but there is no HTTP port exposed.

If the fragment is extended to include "http" as follows:

src/main/jkube/service.yml:
```
spec:
  ports:
    - name: http
      protocol: TCP
      port: 80
      targetPort: 8080
    - name: hazelcast
      protocol: TCP
      port: 5701
      targetPort: 5701
```

then openshift-maven-plugin fails (silently) to generate anything in target/jkube/applyJson, possibly due to name collision for two unnamed service port entries, and deployment subsequenly fails with:

```
[ERROR] oc: Failed to update Service from openshift.yml.
```

since openshift.yml is absent.


### Info

```
OpenShift: 4.6
Platform: Windows 10
Eclipse JKube version : 1.0.2

> mvn -v
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\dev\apache-maven-3.6.3\bin\..
Java version: 15.0.1, vendor: AdoptOpenJDK, runtime: C:\Program Files\AdoptOpenJDK\jdk-15.0.1.9-openj9
Default locale: en_GB, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

Potentially relevant maven properties include:

```
<version.jkube>1.0.2</version.jkube>
<maven.compiler.source>11</maven.compiler.source>
<maven.compiler.target>11</maven.compiler.target>
<kubernetes.trust.certificates>true</kubernetes.trust.certificates>
<jkube.enricher.jkube-service.type>ClusterIP</jkube.enricher.jkube-service.type>
<jkube.enricher.jkube-service.normalizePort>true</jkube.enricher.jkube-service.normalizePort>
<jkube.generator.alias>${project.artifactId}</jkube.generator.alias>
```

I will endeavour to add a simple reproducer project as soon as possible.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -37,6 +37,7 @@ Usage:
 * Fix #509: Port of ServiceDiscoveryEnricher from FMP
 * Fix #511: Namespace as resource fragment results in NullPointerException
 * Fix #521: NPE on Buildconfig#getContextDir if `<dockerFile>` references a file with no directory
+* Fix #513: openshift-maven-plugin: service.yml fragment with ports creates service with unnamed port mapping
 
 ### 1.0.2 (2020-10-30)
 * Fix #429: Added quickstart for Micronaut framework

--- a/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricher.java
+++ b/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricher.java
@@ -26,6 +26,7 @@ import io.fabric8.kubernetes.api.model.ServicePortBuilder;
 import io.fabric8.kubernetes.api.model.ServiceSpec;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
 import org.eclipse.jkube.kit.config.image.ImageConfiguration;
 import org.eclipse.jkube.kit.common.Configs;
 import org.eclipse.jkube.kit.common.util.JKubeProjectUtil;
@@ -529,7 +530,7 @@ public class DefaultServiceEnricher extends BaseEnricher {
 
     private String getDefaultServiceName(Service defaultService) {
         String defaultServiceName = KubernetesHelper.getName(defaultService);
-        if (defaultServiceName != null && !defaultServiceName.isEmpty()) {
+        if (StringUtils.isBlank(defaultServiceName)) {
             defaultServiceName = getContext().getGav().getSanitizedArtifactId();
         }
         return defaultServiceName;
@@ -562,7 +563,7 @@ public class DefaultServiceEnricher extends BaseEnricher {
 
     private String ensureServiceName(ObjectMeta serviceMetadata, ServiceBuilder service, String defaultServiceName) {
         String serviceName = KubernetesHelper.getName(serviceMetadata);
-        if (serviceName != null && !serviceName.isEmpty()) {
+        if (StringUtils.isBlank(serviceName)) {
             service.buildMetadata().setName(defaultServiceName);
             serviceName = KubernetesHelper.getName(service.buildMetadata());
         }
@@ -602,14 +603,14 @@ public class DefaultServiceEnricher extends BaseEnricher {
     }
 
     private void ensurePortName(ServicePort port, String protocol) {
-        if (port.getName() == null || port.getName().isEmpty()) {
+        if (StringUtils.isBlank(port.getName())) {
             port.setName(getDefaultPortName(port.getPort(), getProtocol(protocol)));
         }
     }
 
     private String ensureProtocol(ServicePort port) {
         String protocol = port.getProtocol();
-        if (protocol != null && !protocol.isEmpty()) {
+        if (StringUtils.isBlank(protocol)) {
             port.setProtocol("TCP");
             return "TCP";
         }

--- a/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricher.java
+++ b/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricher.java
@@ -602,7 +602,7 @@ public class DefaultServiceEnricher extends BaseEnricher {
     }
 
     private void ensurePortName(ServicePort port, String protocol) {
-        if (port.getName() != null && !port.getName().isEmpty()) {
+        if (port.getName() == null || port.getName().isEmpty()) {
             port.setName(getDefaultPortName(port.getPort(), getProtocol(protocol)));
         }
     }

--- a/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricherAddMissingPartsTest.java
+++ b/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricherAddMissingPartsTest.java
@@ -85,7 +85,7 @@ public class DefaultServiceEnricherAddMissingPartsTest {
     imageConfigurationWithPort("80");
     final KubernetesListBuilder klb = new KubernetesListBuilder().addToItems(
         new ServiceBuilder().editOrNewSpec().addNewPort()
-            .withName("iana-replaced").withProtocol("TCP").withPort(1337).endPort().endSpec().build());
+            .withProtocol("TCP").withPort(1337).endPort().endSpec().build());
     // When
     enricher.create(null, klb);
     // Then

